client/landing/stepper/declarative-flow/internals/hooks/use-launchpad-decider/index.tsx

```diff
@@ -22,42 +24,37 @@ interface SiteProps {
 }
 
 export const useLaunchpadDecider = ( { exitFlow, navigate }: Props ) => {
-	// placeholder field for the experiment assignment
-	const showCustomerHome = false;
-	const sessionId = getSessionId();
-	let launchpadStateOnSkip: null | 'skipped' = null;
-	if ( showCustomerHome ) {
-		launchpadStateOnSkip = 'skipped';
+	const { getIntent } = useSelect( ( select ) => select( ONBOARD_STORE ) as OnboardSelect, [] );
+	const intent = getIntent();
+	const site = useSite();
+	// site intent is not set until we exit site setup flow so we set it here with onboarding store
+	// as it is used in shouldShowLaunchpadFirst
+	if ( site && site.options && site.options.site_intent === '' ) {
+		site.options.site_intent = intent;
 	}
-
-	const setLaunchpadSkipState = ( siteIdOrSlug: string | number | null ) => {
-		if ( siteIdOrSlug && launchpadStateOnSkip ) {
-			updateLaunchpadSettings( siteIdOrSlug, { launchpad_screen: launchpadStateOnSkip } );
-		}
-	};
+	const showCustomerHome = site && shouldShowLaunchpadFirst( site );
+	const sessionId = getSessionId();
 
 	return {
 		getPostFlowUrl: ( { flow, siteId, siteSlug }: PostFlowUrlProps ) => {
 			if ( showCustomerHome ) {
 				return `/home/${ siteSlug || siteId }`;
 			}
 
-			return addQueryArgs( `/setup/${ flow }/launchpad`, { siteSlug, siteId, sessionId } );
+			return addQueryArgs( `/setup/${ flow }/launchpad`, {
+				siteSlug,
+				siteId,
+				sessionId,
+				showLaunchpad: true,
```

Comment:
I don't see any code that does anything in response to the `showLaunchpad=true` URL parameter. Could you confirm why we're adding it to this code?

Reply:
Yeah I'm not seeing where this query param is still used. I'll remove it.